echo-pubsub/src/main/java/com/netflix/spinnaker/echo/pubsub/amazon/SQSSubscriber.java

```diff
@@ -0,0 +1,267 @@
+        registry.counter(getProcessedMetricId(subscriptionName())).increment();
+      });
+    }
+  }
+
+  private void handleMessage(Message message) {
+    String messageId = message.getMessageId();
+    String messagePayload = unmarshallMessageBody(message.getBody());
+
+    Map<String, String> stringifiedMessageAttributes = message.getMessageAttributes().entrySet().stream()
+      .collect(Collectors.toMap(Map.Entry::getKey, e -> String.valueOf(e.getValue())));
+
+    log.debug("Received Amazon sqs message: {} with payload: {}\n and attributes: {}", messageId, messagePayload, stringifiedMessageAttributes);
+
+    MessageDescription description = MessageDescription.builder()
+      .subscriptionName(subscriptionName())
+      .messagePayload(messagePayload)
+      .messageAttributes(stringifiedMessageAttributes)
+      .pubsubSystem(pubsubSystem)
+      .ackDeadlineMillis(5 * TimeUnit.SECONDS.toMillis(10)) // Set a high upper bound on message processing time.
+      .retentionDeadlineMillis(TimeUnit.DAYS.toMillis(7)) // Expire key after max retention time, which is 7 days.
+      .build();
+
+    AmazonMessageAcknowledger acknowledger = new AmazonMessageAcknowledger(amazonSQS, queueId, message);
+
+    if (subscription.getMessageFormat() != AmazonPubsubProperties.MessageFormat.NONE) {
+      description.setArtifacts(parseArtifacts(description.getMessagePayload(), messageId));
+    }
+    pubsubMessageHandler.handleMessage(description, acknowledger, identity.getIdentity(), messageId);
+  }
+
+  private List<Artifact> parseArtifacts(String messagePayload, String messageId){
+    List<Artifact> artifacts = messageArtifactTranslator.parseArtifacts(messagePayload);
+    if (artifacts == null) {
+      log.debug("No artifacts were found for subscription: {} and messageId: {}", subscription.getName(), messageId);
+    } else {
+      artifacts.forEach( a -> a.setArtifactAccount(subscription.getAccountName()));
+      log.debug(
+        "Artifacts found for subscription: {}: {}",
+        subscription.getName(),
+        String.join(", ", artifacts.stream().map(Artifact::toString).collect(Collectors.toList()))
+      );
+    }
+    return artifacts;
+  }
+
+  private String unmarshallMessageBody(String messageBody) {
+    String messagePayload = messageBody;
+    try {
+      NotificationMessageWrapper wrapper = objectMapper.readValue(messagePayload, NotificationMessageWrapper.class);
+      if (wrapper != null && wrapper.getMessage() != null) {
+        messagePayload = wrapper.getMessage();
+      }
+    } catch (IOException e) {
+      // Try to unwrap a notification message; if that doesn't work,
+      // we're dealing with a message we can't parse. The template or
+      // the pipeline potentially knows how to deal with it.
+      log.error("Unable unmarshal NotificationMessageWrapper. Unknown message type. (body: {})", messageBody, e);
+      registry.counter(getNotAmazonNotificationMetricId(subscriptionName())).increment();
```

Comment:
Not sure what this metric is meant to tell us? The `NotAmazonNotification` sounds a little strange?

Reply:
It does sound strange. This is meant to track any message delivered to the queue that is not from SNS (message type != Notification). I wanted to see if that's something that will happen, and track this, so that I can work in support for parsing messages from those other sources (like SQS). 

But, after thinking about it, I don't think this will happen often enough to track a metric on it. Removing.